contentpal/src/main/java/org/dmfs/android/contentpal/rowdata/ForeignId.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.android.contentpal.rowdata;
+
+import android.content.ContentProviderOperation;
+import android.support.annotation.NonNull;
+
+import org.dmfs.android.contentpal.RowData;
+import org.dmfs.android.contentpal.RowSnapshot;
+import org.dmfs.android.contentpal.TransactionContext;
+
+
+/**
+ * {@link RowData} that adds a foreign id referring to the provided row from another table.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class ForeignId<T> implements RowData<T>
```

Comment:
I think `Referring` is a better name as it expresses that this is `RowData` which adds a reference to another row.